CMakeLists.txt

```diff
@@ -109,7 +109,8 @@ set(JPEGXL_DEP_LICENSE_DIR "" CACHE STRING
     "Directory where to search for system dependencies \"copyright\" files.")
 set(JPEGXL_FORCE_NEON false CACHE BOOL
     "Set flags to enable NEON in arm if not enabled by your toolchain.")
-
+set(JPEGXL_VERSION "" CACHE STRING
```

Comment:
What's the idea behind this? The PR description says to provide additional version information but this patch alone doesn't provide additional information. You can already pass `-DJPEGXL_VERSION=123456` to cmake, but I don't think it would be beneficial to cache this value in cmake.

Reply:
Now the `-DJPEGXL_VERSION` flag is ignored when building.

Reply:
ok, that's a different problem. The issue here is the automatic variable JPEGXL_VERSION that cmake creates based on the project(); any other name would work. We should fix that instead.

Reply:
Simply defining `JPEGXL_VERSION` in CMakelists solves the problem for me. But if there is more proper solution, I'd be glad to have it. 😉

Reply:
#321 explains what's the issue here. The more proper solution would be to set the major.minor.patch version at the project line, but I'd need to move around some macros since they are currently only defined in the lib.gni and lib/CMakeLists.txt and we have some tooling to keep them in sync. However, that would define the JPEGXL_VERSION as the version number (like 0.3.7) which is fine, but we also want to have a version string you can include in the tools defined from packages which was the purpose of this macro; so changing the project name to avoid the collision with JPEGXL_VERSION sounds a reasonable first step.

Reply:
Thanks for the explanation. If this PR is not needed, please feel free to close it.

Reply:
Thank you for reporting the issue!